src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, null, null);
+        if (first == null) {
+            first = newNode;
+        } else {
+            newNode.previous = last;
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            first.previous = new Node<>(value, null, first);
+            first = first.previous;
+        } else {
+            Node<T> currentNode = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(value, currentNode.previous, currentNode);
+            currentNode.previous = newNode;
+            newNode.previous.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNodeByIndex(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeToDelete = getNodeByIndex(index);
+        unlink(nodeToDelete);
+        return nodeToDelete.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> nodeToDelete = getNodeByValue(object);
+        if (nodeToDelete == null) {
+            return false;
+        } else {
+            unlink(nodeToDelete);
+            return true;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return first == null;
+    }
+
+    public static class Node<T> {
```

Comment:
```suggestion
   private static class Node<T> {
```

I think we could make it private

Reply:
Fixed